crates/trie/parallel/src/async_root.rs

```diff
@@ -126,8 +124,15 @@ where
                     )
                     .with_prefix_set(prefix_set)
                     .calculate(retain_updates)?)
-                });
-            storage_roots.insert(hashed_address, handle);
+                })();
+
+                (hashed_address, result)
+            })
+            .collect();
```

Comment:
Isn't this now equivalent to the `Parallel` implementation? because this first computes all hashed addresses and only then starts iterating over the trie elements, whereas the previous impl would spawn the jobs and await the results

Reply:
you are totally right, `calculate` was completely sync. now the jobs for storage root calculation are spawned using the global thread pool and we await the results, ptal